flang/lib/Lower/Bridge.cpp

```diff
@@ -588,6 +588,13 @@ class FirConverter : public Fortran::lower::AbstractConverter {
         std::nullopt);
   }
 
+  bool isPresentShallowLookup(Fortran::semantics::Symbol &sym) override final {
+    Fortran::lower::SymbolBox sb = shallowLookupSymbol(sym);
+    if (shallowLookupSymbol(sym))
```

Comment:
Why is this called twice? And shouldn't this function just be 
```
bool isPresentShallowLookup(Fortran::semantics::Symbol &sym) {
  return shallowLookupSymbol(sym)
}
```
And in that case do we need to make this function, or can we directly call this shallowLookup() function at the use site?

Reply:
`shallowLookupSymbol()` does not seem to be exposed to any interface we have access to while lowering. So I added this interface in AbstractConverter. I'll recheck once though.

And will modify `isPresentShallowLookup` is at all it needs to be there.

Reply:
@shraiysh I checked this. `shallowLookupSymbol` is operating upon a symbol map `localSymbols` defined and maintained in `Bridge.cpp`. So in any case, we need to allow an interface to access the same during lowering OpenMP clauses.